### PR TITLE
Multiple updates regarding LLMQs

### PR DESCRIPTION
**Pull request by codablock**
See individual commits. These are mostly updates so that the DIPs match the implementation.

### CODE REVIEW

**Comment by thephez** [comment]
@codablock Can you bump the CbTx version number in https://github.com/dashpay/dips/blob/master/dip-0002-special-transactions.md also?

**Comment by codablock** [comment]
@thephez done

File: dip-0004.md [hunk 1]
```diff
@@ -88,6 +94,23 @@ A SML entry consists of the following fields:
...
+To calculate the merkle root:
+
+  1. Collect all final commitments from all LLMQs which are in the active set at the given block height. All LLMQ types need
+     to be collected.
+  2. Calculate the hash of all these commitments and collect the hashes in a list.
+  3. Sort this list in ascending order by the previously calculated hash.
+  4. Calculate the merkle root from this list of hashes in the same way it is done when calculating the merkle root of the block transactions.
+
+Please refer to [DIP6 - Long-Living Masternode Quorums](https://github.com/dashpay/dips/blob/master/dip-0006.md) for more details on LLMQs
+and final commitments. The final commitment which is referred multiple times in this DIP is the `qfcommit` message found in
```
Comment (thephez) [inline dip-0004.md]: "...which is referred multiple times..." -> "...referred to multiple times..."

File: dip-0004.md [hunk 2]
```diff
@@ -132,6 +165,24 @@ To achieve this, It is suggested to first request a full masternode list from an
...
+## Tracking/Updating and verifying the active LLMQ sets based on MNLISTDIFF
+
+With the introduction of version 2 CbTx special transactions, a merkle root for the currently active LLMQ sets was added
+to the coinbase. This merkle root allows (SPV) clients to maintain the active LLMQ sets without the full chain.
+
+It is suggested to couple the house-/bookkeeping of active LLMQ sets with the masternode list bookkeeping described in
+the previous section and perform the following additional steps:
+
+  1. Create a copy of the active LLMQ sets which were given at "baseBlockHash". If “baseBlockHash” is all-zero, empty sets must be used.
+     As there are multiple LLMQ types.
```
Comment (thephez) [inline dip-0004.md]: Looks like the "As there are multiple LLMQ types" should just be the end of the previous sentence.

Comment (codablock) [reply dip-0004.md]: When rendered, it will actually be part of the previous sentence. I just don't like too long lines in Markdown :)

**Comment by codablock** [comment]
Added a commit to handle a review comment. Neew re-ACK.

File: dip-0004.md (hunk 2 above)
Comment (codablock) [reply dip-0004.md]: Oh...now I see...that sentence shouldn't be there at all, updated last commit.

File: dip-0004.md [hunk 3]
```diff
@@ -132,6 +165,23 @@ To achieve this, It is suggested to first request a full masternode list from an
...
+
+With the introduction of version 2 CbTx special transactions, a merkle root for the currently active LLMQ sets was added
+to the coinbase. This merkle root allows (SPV) clients to maintain the active LLMQ sets without the full chain.
+
+It is suggested to couple the house-/bookkeeping of active LLMQ sets with the masternode list bookkeeping described in
+the previous section and perform the following additional steps:
+
+  1. Create a copy of the active LLMQ sets which were given at "baseBlockHash". If “baseBlockHash” is all-zero, empty sets must be used.
+  2. Delete all entries found in "deletedQuorums" from the corresponding active LLMQ sets.
+  3. Verify each final commitment found in "newQuorums", by the same rules found in [DIP6 - Long-Living Masternode Quorums](https://github.com/dashpay/dips/blob/master/dip-0006.md).
```
Comment (UdjinM6) [inline dip-0004.md]: This step should be dropped. SPVs shouldn't verify this, instead they should rely on PoW+merkleroot (steps 5 and 6) - merkle root won't match if some final commitment was tampered.

**Pull request merged**